overlord/snapstate/conflict.go

```diff
@@ -71,7 +71,7 @@ func AddAffectedSnapsByKind(kind string, f AffectedSnapsFunc) {
 
 func affectedSnaps(t *state.Task) ([]string, error) {
 	// snapstate's own styled tasks
-	if t.Has("snap-setup") || t.Has("snap-setup-task") {
+	if t.Has("snap-setup") || t.Has("snap-setup-task") || t.Has("restore-state") {
```

Comment:
Hmm I think there's some confusing code in snapshots. We have SnapshotManager.affectedSnaps() which registered for affected snaps check via `snapstate.AddAffectedSnapsByAttr("snapshot-setup", manager.affectedSnaps)` and looks for `snapshot-setup` already. In theory this should be enough to cause conflict. Perhaps we just need a managers level test to confirm?

Reply:
This was the first change that I made in this branch, and now I realize that it was not even needed :-)
I reverted it now.